File: Lpp.Dns.DTO/Enums/Race.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;

namespace Lpp.Dns.DTO.Enums
{
    /// <summary>
    /// Types of Race
    /// </summary>
    [DataContract]
    public enum Race
    {
        /// <summary>
        /// Indicates Race is UnKnown
        /// </summary>
        [EnumMember, Description("Unknown")]
        Unknown = 0,
        /// <summary>
        /// Indicates Race is American Indian or Alaska Native
        /// </summary>
        [EnumMember, Description("American Indian or Alaska Native")]
        Native = 1,
        /// <summary>
        /// Indicates Race is Asian
        /// </summary>
        [EnumMember, Description("Asian")]
        Asian = 2,
        /// <summary>
        /// Indicates Race is Black or African American
        /// </summary>
        [EnumMember, Description("Black or African American")]
        Black = 3,
        /// <summary>
        /// Indicates Race is Native Hawaiian or Other Pacific Islander
        /// </summary>
        [EnumMember, Description("Native Hawaiian or Other Pacific Islander (NHOPI)")]
        Pacific = 4,
        /// <summary>
        /// Indicates Race is White
        /// </summary>
        [EnumMember, Description("White")]
        White = 5,
        /// <summary>
        /// Indicates Race is Multiple
        /// </summary>
        [EnumMember, Description("Multiple Race")]
        Multiple = 6,
        /// <summary>
        /// Indicates Race is Refuse to answer
        /// </summary>
        [EnumMember, Description("Refuse to Answer")]
        Refuse = 7,
        /// <summary>
        /// Indicates Race is No Information
        /// </summary>
        [EnumMember, Description("No Information")]
        NI = 8,
        /// <summary>
        /// Indicates Race is Other
        /// </summary>
        [EnumMember, Description("Other")]
        Other = 9
        
    }
}

File: Plugins/Lpp.Dns.General.Metadata/Content/ViewModels/Terms/Race.js
/// <reference path="../../../../../lpp.dns.portal/scripts/common.ts" />
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
///// <reference path="../../Models/Terms.ts" />
/// <reference path="../../Models/Terms/Race.ts" />
///// <reference path="../Terms.ts" />
var DataCheckerViewModels;
(function (DataCheckerViewModels) {
    var RaceTerm = (function (_super) {
        __extends(RaceTerm, _super);
        function RaceTerm(raceData) {
            var _this = _super.call(this, RequestCriteriaModels.TermTypes.RaceTerm) || this;
            var dummy = [];
            _this.Races = ko.observableArray(raceData ? raceData.Races : dummy);
            _super.prototype.subscribeObservables.call(_this);
            return _this;
        }
        RaceTerm.prototype.toData = function () {
            var superdata = _super.prototype.toData.call(this);
            var raceData = {
                TermType: superdata.TermType,
                Races: this.Races()
            };
            return raceData;
        };
        return RaceTerm;
    }(RequestCriteriaViewModels.Term));
    RaceTerm.RacesList = [
        new Dns.KeyValuePairData('Unknown', DataCheckerModels.RaceTypes.Unknown),
        new Dns.KeyValuePairData('American Indian/Alaska Native', DataCheckerModels.RaceTypes.AmericanIndianOrAlaskaNative),
        new Dns.KeyValuePairData('Asian', DataCheckerModels.RaceTypes.Asian),
        new Dns.KeyValuePairData('Black/African American', DataCheckerModels.RaceTypes.BlackOrAfricanAmerican),
        new Dns.KeyValuePairData('Native Hawaiian/Pacific Islander', DataCheckerModels.RaceTypes.NativeHawaiianOrOtherPacificIslander),
        new Dns.KeyValuePairData('White', DataCheckerModels.RaceTypes.White),
        new Dns.KeyValuePairData('Missing', DataCheckerModels.RaceTypes.Missing)
    ];
    DataCheckerViewModels.RaceTerm = RaceTerm;
})(DataCheckerViewModels || (DataCheckerViewModels = {}));
